### PR TITLE
Management cluster creation without clusterctl

### LINKED ISSUE #462: Create a management cluster without using clusterctl
/kind documentation

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]
Our project allows us to create a management cluster following simple steps, and without the use of clusterctl, documenting that in the repo will be useful for future contributors and users.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]
The management cluster will be created using the makefile by the command `make create-management-cluster`
I guess we can do the same for workload cluster as well, if I am able do that will add the step in the same doc.

### PR DESCRIPTION
**Pull request by sayantani11**
**What type of PR is this?**

**What this PR does / why we need it**:
Doc that includes steps for creating a management cluster without clusterctl.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #462 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Create a management cluster using clusterctl
```

### CODE REVIEW

File: docs/book/src/developers/cluster-creation.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Creating cluster without clusterctl
+
+This document describes how to create a management cluster and workload cluster without using clusterctl.
+For creating a cluster with clusterctl, checkout our [Cluster API Quick Start](https://cluster-api.sigs.k8s.io/user/quick-start.html)
+
+## For creating a Management cluster
+
+1. Build required images by using the following commands:
+   
+   - `docker pull gcr.io/k8s-staging-cluster-api-gcp/cluster-api-gcp-controller:v20211110-v0.4.0-18-gba203a6a`
```
Comment (cpanato) [inline docs/book/src/developers/cluster-creation.md:10]: why you pull this image? i guess this is not required

Comment (sayantani11) [reply docs/book/src/developers/cluster-creation.md:10]: I did this for the controller image actually... I will remove this

**Review by cpanato — APPROVED** [review]
/approve
/lgtm

thanks!

**Pull request merged**